Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -0,0 +1,619 @@
+
+                // Offset from beginning of packet (instances + current offset)
+                if (dataBufferOffset !== (numInstances * dataInstanceSize + offset)) {
+                    throw new RuntimeError('Invalid dataBufferOffset');
+                }
+
+                // Verify the packets is all there header + instances + dataBuffer + metaBuffer
+                if (dataBufferOffset + dataBufferSize + metaBufferSize !== uncompressedPacket.length) {
+                    throw new RuntimeError('Invalid packet offsets');
+                }
+
+                // Read all the instances
+                var instances = [];
+                for (var i = 0; i < numInstances; ++i) {
+                    var bitfield = dv.getUint8(offset);
+                    ++offset;
+
+                    ++offset; // 2 byte align
+
+                    var cnodeVersion = dv.getUint16(offset, true);
+                    offset += sizeOfUint16;
+
+                    var imageVersion = dv.getUint16(offset, true);
+                    offset += sizeOfUint16;
+
+                    var terrainVersion = dv.getUint16(offset, true);
+                    offset += sizeOfUint16;
+
+                    // Number of channels stored in the dataBuffer
+                    //var numChannels = dv.getUint16(offset, true);
+                    offset += sizeOfUint16;
+
+                    offset += sizeOfUint16; // 4 byte align
+
+                    // Channel type offset into dataBuffer
+                    //var typeOffset = dv.getInt32(offset, true);
+                    offset += sizeOfInt32;
+
+                    // Channel version offset into dataBuffer
+                    //var versionOffset = dv.getInt32(offset, true);
+                    offset += sizeOfInt32;
+
+                    offset += 8; // Ignore image neighbors for now
+
+                    // Data providers aren't used
+                    ++offset; // Image provider
+                    ++offset; // Terrain provider
+                    offset += sizeOfUint16; // 4 byte align
+
+                    instances.push(new TileInformation(bitfield, cnodeVersion,
+                        imageVersion, terrainVersion));
+                }
+
+                var tileInfo = that._tileInfo;
+                var index = 0;
+
+                function populateTiles(parentKey, parent, level) {
+                    var isLeaf = false;
+                    if (level === 4) {
+                        if (parent.hasSubtree()) {
+                            return; // We have a subtree, so just return
+                        }
+
+                        isLeaf = true; // No subtree, so set all children to null
+                    }
+                    for (var i = 0; i < 4; ++i) {
+                        var childKey = parentKey + i.toString();
+                        if (isLeaf) {
+                            // No subtree so set all children to null
+                            tileInfo[childKey] = null;
```

Comment:
They mean different things. If it's undefined then we haven't retrieved the child yet, if it's null the parent has told us it doesn't exist.